src/System.Private.CoreLib/src/System/Threading/ThreadPool.CoreRT.cs

```diff
@@ -0,0 +1,85 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+/*=============================================================================
+**
+**
+**
+** Purpose: Class for creating and managing a threadpool
+**
+**
+=============================================================================*/
+
+#pragma warning disable 0420
+
+using Internal.Runtime.Augments;
+using Internal.Runtime.CompilerServices;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+
+using Thread = Internal.Runtime.Augments.RuntimeThread;
+
+namespace System.Threading
+{
+    internal sealed partial class ThreadPoolWorkQueue
+    {
+        internal static bool DispatchWithExceptionHandling()
```

Comment:
We do not need this try/catch/FailFast. The places that call DispatchWithExceptionHandling directly. The unhandled exception will be dealt with by the next frame in both cases. It should be actually a better experience since it will be a proper unhandled exception and not a fail fast.

Reply:
Are you sure about it? Specifically the one in ThreadPool.Windows.cs felt like it would fall through the Win32 API callback and I was not sure how that would behave.

Reply:
It will turn into a regular unhandled exception when it falls through.